tests/test_domain_js.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+"""
+    test_domain_js
+    ~~~~~~~~~~~~~~
+
+    Tests the JavaScript Domain
+
+    :copyright: Copyright 2007-2016 by the Sphinx team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+import pytest
+
+
+@pytest.mark.sphinx(testroot='domain-js')
+def test_build_domain_py_xrefs_resolve_correctly(app, status, warning):
+    from sphinx.domains.javascript import JavaScriptDomain
+
+    calls = {}
+
+    def wrapped_find_obj(fn):
+        def wrapped(*args):
+            ret = fn(*args)
+            # args = [domain, env, env object, role text, role type, order]
+            calls[args[2:-1]] = ret
+            return ret
+        return wrapped
```

Comment:
Also, this could perhaps be simplified to call `app.env.domains['js'].find_obj` instead of mocking.